prebuilt-ui/rtp-tracks-compositing/public/index.html

```diff
@@ -0,0 +1,207 @@
+      		},
+      	});
+
+      	call.on('recording-started', (e) => {
+      		console.log(
+      			'Recording started, id: ',
+      			e.recordingId,
+      			', Instructor session: ',
+      			window.instructorSessionId
+      		);
+      		recordingId = e.recordingId;
+      	});
+
+      	call.on('recording-stopped', (e) => {
+      		console.log('Recording stopped');
+      	});
+
+      	call.on('left-meeting', (e) => {
+      		// might want to just start processing the recording after the meeting starts in case the instructor closes the window
+      		// or use webhook-meeting-end to process recording
+      		console.log('Leaving meeting');
+      	});
+
+      	window.token = await createMeetingToken({
+      		enable_recording: 'rtp-tracks',
+      		user_name: 'Instructor',
+      		user_id: 'instructor',
+      		is_owner: true,
+      	});
+
+      	let participants = await call.join({
+      		token: token
+      	});
+
+      	call.setLocalVideo(true);
+      	call.setActiveSpeakerMode(false);
+
+      	window.instructorSessionId = participants.local.session_id;
+      }
+
+      async function createMeetingToken(properties) {
+      	let response = await fetch('/api/daily/meeting-tokens', {
+      		method: 'post',
+      		body: JSON.stringify({
+      			properties
+      		}),
+      	});
+      	return (await response.json()).token;
+      }
+
+      async function startSession() {
+      	call.sendAppMessage({
+      		message: 'startSession'
+      	}, '*');
+      	call.startRecording();
+      }
+
+      async function stopSession() {
+      	call.sendAppMessage({
+      		message: 'stopSession'
+      	}, '*');
+      	call.stopRecording();
+      	console.log('Starting processing for recording id:', recordingId);
+      	processRecording(￼)
+      }
+
+      // recording post-processing code that could be moved server-side so
+      // you don't have to leave a browser window open
+
+      var tracks;
+
+      async function wait(ms = 1000) {
+      	return new Promise(resolve => {
+      		setTimeout(resolve, ms);
+      	});
+      }
+
+      async function getRecordingInfo(recordingId) {
+      	let response = await fetch(
+      		"/api/daily/recordings/" + recordingId, {
+      			method: "get",
+      		}
+      	);
+      	// todo: a cleaner way to get the most recent recording in the room?
+      	//console.log("response: ", (await response.text()))
+      	return await response.json();
+      }
+
+      async function awaitTracks(recordingId) {
+      	while (true) {
+      		let rec = await getRecordingInfo(recordingId);
+      		// console.log("Recording info: ", rec);
+      		if (
+      			rec.tracks.find(e => e.type == "audio" && e.user_id == "instructor") &&
+      			rec.tracks.find(e => e.type == "video" && e.user_id == "instructor")
+      		) {
+      			return rec.tracks;
+      		} else {
+      			console.log("Waiting for tracks, currently: ", rec.tracks)
+      			await wait();
```

Comment:
is this some sort of hack to do long polling of that endpoint?

Reply:
Yes... once the call is over, we have to wait for the necessary tracks to appear in the track list, so we have to poll